### PR TITLE
Adapt new ESLint and Prettier configuration

### PR DESCRIPTION
**Pull request by medikoo**
It covers following tasks, covered with individual commits:
- https://github.com/serverless/serverless/pull/6284/commits/346faf73f5cd06cf81d9fb137d1d7b4c473c6f05 - New ESlint & Prettier config integration
- https://github.com/serverless/serverless/pull/6284/commits/72ba2326050af9e833032353ef16c7974210161c - Travis config fix and cleanup:
   - Ensure to take some git history, to avoid issues where [tested commit is not found](https://travis-ci.org/serverless/serverless/jobs/548129914)
   - Separate PR and branch lint/test job configuration
- https://github.com/serverless/serverless/pull/6284/commits/4e2ed4dd8297561cf575394450b83050758d1d09 - Confirm changed files against Prettier in CI
- https://github.com/serverless/serverless/pull/6284/commits/f98b90ef2f9b01f77d0176c63f72b53d6f0a1c3c - Configure `test-ci` a command that runs all validation checks that are pursued in CI
- https://github.com/serverless/serverless/pull/6284/commits/81b419bad5ee0e24bd07ad96c6a999a2058fce3b - Fix .`editorconfig` configuration for markdown files (so trim of ending whitespace is not pursued)
- https://github.com/serverless/serverless/pull/6284/commits/2a870d025147e18572536c5a5ddea6a5c1cd3725 - Educate on pursued validation checks in PR template
- https://github.com/serverless/serverless/pull/6284/commits/f46dc0321d7627d10c341ed792f28d1487073dea - Add `.js` extension to `bin/serverless` so it's properly covered by ESLint and Prettier
- https://github.com/serverless/serverless/pull/6284/commits/4944f471b14bd9398ea9c05a35d81b7f2eb62b02 - (automatic) Prettify all files
- https://github.com/serverless/serverless/pull/6284/commits/e6ae8983de198e004ac5c166ccc8ee44315cfa12 - Fix lint issues surfaced after automatic prettification

__Compare all changes aside of automatic prettification: https://github.com/serverless/serverless/compare/master...583197adfe8afa54f408d68595baa83660e7828e__

--- 

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,12 +46,13 @@
   ],
   "main": "lib/Serverless.js",
   "bin": {
-    "serverless": "./bin/serverless",
-    "slss": "./bin/serverless",
-    "sls": "./bin/serverless"
+    "serverless": "./bin/serverless.js",
```
Comment (pmuens) [inline package.json]: Might not be a huge difference, but why don't we stick to the `./bin/serverless` file instead?

If we're using the `.js` file here, what is the `./bin/serverless` file used for?

Comment (medikoo) [reply package.json]: > Might not be a huge difference, but why don't we stick to the ./bin/serverless file instead?

File type discoverability is more complicated for many processors, e.g. ESLint, Prettier won't pick it and that means we're not guarded against errors as we should be.

I think it's a good reason to have extension on binary, and most projects I know switched to that (you may check: ESLint, Webpack, Prettier, npm etc.)

> If we're using the .js file here, what is the ./bin/serverless file used for?

It's kept to not accidentally break things for those which by chance refer to bin/serverless directly.

It's to be removed with next breaking version release. I explained that in a file: 

https://github.com/serverless/serverless/blob/bc128d5d269ce2227271491e211cade6f16f83dc/bin/serverless#L3-L5

**Comment by medikoo** [comment]
> I'd say it's GTM, but I'm concerned by all the files re-formattings which will result in a lot of conflicts. There's no perfect timing to introduce this however. Maybe we can discuss this in our meeting today.

Sure, the alternative I see. Is that we do not validate prettier, but we allow to work with prettier if one wants to, but that would result with constant white-space noise in PR changes and git history, which seems quite dirty, I don't think I like that approach.

As Prettier adoption is really huge now, I wouldn't be much afraid of that change, and if developer of some existing PR struggles with conflicts, we may also help (I definitely can, if needed)

File: package.json (hunk 1 above)
Comment (pmuens) [reply package.json]: > File type discoverability is more complicated for many processors, e.g. ESLint, Prettier won't pick it and that means we're not guarded against errors as we should be.

Yep, I'm in the same boat. Was just wondering why we have such files but still keep the old one. However your comment regarding potential breaking changes makes sense (haven't seen the comment).

**Comment by pmuens** [comment]
> Sure, the alternative I see. Is that we do not validate prettier, but we allow to work with prettier if one wants to, but that would result with constant white-space noise in PR changes and git history, which seems quite dirty, I don't think I like that approach.

> As Prettier adoption is really huge now, I wouldn't be much afraid of that change, and if developer of some existing PR struggles with conflicts, we may also help (I definitely can, if needed)

Yes, I can see that and I'm in for making the change now (given that it's trivial to prettify files nowadays) rather than having this loose "contract" which doesn't enforce prettified files.

**Pull request merged**